### ISSUE TITLE
CopyAcross Agent into blank fields

### ISSUE THREAD

**Issue by cjconroy** [ArctosDB/arctos#7746]
A lot of specimens have distinct collectors and preparators. In the way my data entry screen is set up, I enter an agent as the first agent, then the preparator as the second agent. It is handy to use the CopyAcross button to then put the preparator name in subsequent fields, like all the measurements. However, it also puts it into agent 1, the collector. A fix on my end would be to enter the preparator first, then copyacross, then enter collector. But, could copy across only work on either blank agents, or agents other than agent 1?

**Comment by mkoo** [comment]
Seems like a solid request: CopyAcross Agent value only fills in blank Agent fields
Being the default action makes sense since logically one starts with a blank entry form or one starts wtih a few names already in place which you dont want changed

**Comment by dustymc** [comment]
"CopyAcross only fills in blank fields" is actionable, but probably entirely incompatible with some outcomes of https://github.com/ArctosDB/arctos/issues/8456 - this would result in filling 80 part_nn_attribute_determiner_m fields which would then have to be removed or used before the record could load, for example.

**Comment by mkoo** [comment]
What is the action now? It fills in Agent only for the fields viewable. It doesnt seem to fill in ALL 80 parts....

**Comment by cjconroy** [comment]
I don't know if the order of things is the same for everyone, but for example, if I have a second agent in the agents box and CopyAcross, it will fill in all the agents in the agent box and below, but will not change the agent in Identification 1. The frustration is that agent 1 is usually the collector and agent 2 the prepper who measured the specimen, etc. So, it is very handy to copyacross to fill those fields. I got around this in a simple accession recently by just putting the agent names into every field in the customize so I wouldn't have to copy across. But, not all accessions are so simple.

**Comment by dustymc** [comment]
> It fills in Agent only for the fields viewable. 

Correct, and I think that makes sense when things like not providing a part result in the attribute determiners for that part being ignored. https://github.com/ArctosDB/arctos/issues/8456 would mean that the one click would **cost** multiple click-select-backspace operations unless **every** Agent will be used - which I think would be nearly never and the button would just be a nuisance in that environment.
